template/addons/next-auth/middleware.ts

```diff
@@ -0,0 +1,20 @@
+// Example of a restricted route that only authenticated users can access https://next-auth.js.org/configuration/nextjs#middleware
+
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import { getToken } from "next-auth/jwt";
+
+export async function middleware(request: NextRequest) {
```

Comment:
Should we not use the [default middleware](https://next-auth.js.org/configuration/nextjs#basic-usage) prescribed by NextAuth.js?

Reply:
It depends. That middleware will automatically redirect you to your signin page. The one I added takes you to the homepage. If you think that is correct I can replace it with the one from the docs

Reply:
Yea I guess and that would be a 404 page...

Having a custom one seems over the top for a scaffold, but at the same time it better showcases how to use NextAuth so I think you are right in keeping it like this. Good call!